### ISSUE TITLE
Can create/remove objects but content is lost

### ISSUE THREAD

**Issue by benkehoe** [s3fs-fuse/s3fs-fuse#141]
Starting with commit https://github.com/s3fs-fuse/s3fs-fuse/commit/114966e7c0371eed17e2898cdf505ba4dbfb9b59 (I have narrowed it to this commit) I observe the following:
```
$ s3fs my_bucket ~/s3data && cd ~/s3data
$ echo "foo" > bar
bash: echo: write error: Input/output error
$ ls
bar
$ cat bar
$ rm bar
$ ls
$
```
The object can be seen to be present in the S3 console, but with a size of 0 bytes.

**Comment by andrewgaul** [comment]
This fails in `small-integration-test.sh` as well.  We should really automate these tests per-pull request via Travis #129!

**Comment by admorphit** [comment]
+1

Am using centos 7 and getting error also
```
s3fs mybucket -f -d -o allow_other,del_cache,endpoint=us-west-1,url=https://s3-us-west-1.amazonaws.com /mnt/s3data/
```
Copy command fails
```
cp s3data/bigfile.mp4 /home
cp: error reading ‘s3data/bigfile.mp4’: Input/output error
cp: failed to extend ‘/home/bigfile.mp4’: Input/output error
```
and here's the log
```
s3fs_getattr(721): [path=/bigfile.mp4]
    HeadRequest(2112): [tpath=/bigfile.mp4]
    RequestPerform(1572): connecting to URL https://mybucket.s3-us-west-1.amazonaws.com/bigfile.mp4
    RequestPerform(1588): HTTP response code 200
    AddStat(248): add stat cache entry[path=/bigfile.mp4]
    GetStat(171): stat cache hit [path=/bigfile.mp4][time=1425723772][hit count=0]
s3fs_open(1911): [path=/bigfile.mp4][flags=32768]
    DelStat(370): delete stat cache entry[path=/bigfile.mp4]
    HeadRequest(2112): [tpath=/bigfile.mp4]
    RequestPerform(1572): connecting to URL https://mybucket.s3-us-west-1.amazonaws.com/bigfile.mp4
    RequestPerform(1588): HTTP response code 200
    AddStat(248): add stat cache entry[path=/bigfile.mp4]
    GetStat(171): stat cache hit [path=/bigfile.mp4][time=1425723772][hit count=0]
    ParallelGetObjectRequest(1105): [tpath=/bigfile.mp4][fd=5]
    GetStat(171): stat cache hit [path=/bigfile.mp4][time=1425723772][hit count=1]
    PreGetObjectRequest(2417): [tpath=/bigfile.mp4][start=0][size=10485760]
    PreGetObjectRequest(2417): [tpath=/bigfile.mp4][start=10485760][size=10485760]
    PreGetObjectRequest(2417): [tpath=/bigfile.mp4][start=20971520][size=10485760]
    PreGetObjectRequest(2417): [tpath=/bigfile.mp4][start=31457280][size=10485760]
    PreGetObjectRequest(2417): [tpath=/bigfile.mp4][start=41943040][size=10485760]
    Request(3627): [count=5]
s3fs_read(1957): could not find opened fd(/bigfile.mp4)
s3fs_read(1957): could not find opened fd(/bigfile.mp4)
s3fs_read(1957): could not find opened fd(/bigfile.mp4)
s3fs_read(1957): could not find opened fd(/bigfile.mp4)
s3fs_read(1957): could not find opened fd(/bigfile.mp4)
s3fs_flush(2017): [path=/bigfile.mp4][fd=5]
    GetStat(171): stat cache hit [path=/bigfile.mp4][time=1425723772][hit count=2]
s3fs_release(2057): [path=/bigfile.mp4][fd=5]
s3fs_release(2071): could not find fd(file=/bigfile.mp4)
```
Thoughts?

**Comment by admorphit** [comment]
Managed to resolve the issue today by upgrading fuse from 2.9.2 to 2.9.3. Am using centos 7.

Hope this helps someone.

**Comment by ggtakec** [comment]
Hi,
I confirmed that this is a bug by commit 114966e.
I will fix this problem as soon as possible.
Please wait a while.

**Comment by ggtakec** [comment]
I fixed and merged master branch at #143.
Please try to check it.

Thanks in advance for your help.

**Comment by benkehoe** [comment]
Looks fixed to me, and I'm still on fuse 2.9.2.
